### PR TITLE
Fixes for Big Sur and M1

### PR DESCRIPTION
**Pull request by nevyn**
Some tasks need to be taken care of:

- [x] (Big Sur) Remove image_base to fix luajit linking issue
- [x] (M1) Fix broken [glfw create window](https://github.com/glfw/glfw/issues/1809)

### CODE REVIEW

**Comment by shakesoda** [comment]
does GC64 completely obviate the image_base stuff, or just on arm64?

**Comment by nevyn** [comment]
@shakesoda GC64 doesn't work on 32-bit builds, but it works on x86_64 and so on. I noticed CI switched to 64-bit the other day, so perhaps we're getting rid of 32-bit builds for lovr, in which case we're fine.

**Comment by nevyn** [comment]
According to @Voxar, OpenAL-Soft doesn't work on Apple Silicon 😅 so, good timing with the mini audio rewrite!

**Comment by nevyn** [comment]
okay so glfw is broken on M1... investigating (see https://github.com/glfw/glfw/issues/1809)

**Comment by shakesoda** [comment]
@nevyn why wouldn't openal work? some deprecated calls being made still or something? i had to fix a couple of those in stepmania the other day, but otherwise audio seemed the same as usual.

**Comment by nevyn** [comment]
@shakesoda hm can't find the link, maybe that's a false alarm :S We'll dig into that once we get past the broken graphics

**Comment by nevyn** [comment]
Waiting for https://github.com/glfw/glfw/pull/1833 to get merged upstream, then I can bump glfw in lovr.

**Comment by nevyn** [comment]
Audio works fine! After the GLFW patch, I think this PR is done :)

File: CMakeLists.txt [hunk 1]
```diff
@@ -115,9 +115,10 @@ endif()
 
 # Lua
 if(LOVR_USE_LUAJIT AND NOT EMSCRIPTEN)
-  if (APPLE)
-    set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -pagezero_size 10000 -image_base 100000000")
-  endif()
+  # Required to make luajit compile without breaking linker cache on macOS Big Sur. Also, is a better,
+  # newer GC so enable it on all builds.
+  set(LUAJIT_ENABLE_GC64 ON)
```
Comment (bjornbytes) [inline CMakeLists.txt:120]: Yes!!

**Comment by nevyn** [comment]
@bjornbytes The maintainer for GLFW seems to be out for a while; want to wait for it to go through, or switch to alloverse or some other fork?

Also, it'd be nice to add arm64 support to the Mac binary so it's universal intel-arm; how is CI configured?

**Comment by bjornbytes** [comment]
We can wait for it to get merged to upstream GLFW.  Is it alright if I merge the LuaJIT fix here in the meantime?

There is no macOS CI atm but I am planning to move from appveyor to GitHub Actions and add more CI platforms.

**Comment by nevyn** [comment]
> We can wait for it to get merged to upstream GLFW. Is it alright if I merge the LuaJIT fix here in the meantime?

Absolutely, sure thing!

**Comment by ItsElmo** [comment]
So is there a fix this is very confusing

**Comment by bjornbytes** [comment]
I can confirm that Big Sur is working now, I don't know about M1

**Pull request merged**